### PR TITLE
PsrAutoloadingFixer - do not remove directory structure from the Class name

### PR DESCRIPTION
**Pull request by kubawerlos**
Fixes https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/5377

Pinging @jaydiablo for review (as a reporter of the bug) - the issue you have reported is the one in "class with wrong casing (2 levels namespace)" case.

### CODE REVIEW

File: tests/Fixer/Basic/Psr4FixerTest.php [hunk 1]
```diff
@@ -47,7 +47,7 @@ class bar {}
 
         $expected = <<<'EOF'
 <?php
-class Bar {}
+class Psr4_Foo_Bar {}
```
Comment (kubawerlos) [inline tests/Fixer/Basic/Psr4FixerTest.php:50]: This is a revert of a change from https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5348#discussion_r539700027

Comment (GrahamCampbell) [reply tests/Fixer/Basic/Psr4FixerTest.php:50]: I don't think PSR4 permits having a class name like that. Only PSR0?

Comment (kubawerlos) [reply tests/Fixer/Basic/Psr4FixerTest.php:50]: I guess it should not touch the original name `Psr4_fOo_bAr` at all right? So neither the previous or the new version is correct.

The more I'm thinking (reading/writing code) for the PSR0/4 fixer the more I think the common fixer was a bad idea.

PSR-0 fixer could stay with the `dir` config value and PSR-4 one could some config to fully support PSR-4 which is changing the whole namespace to the expected one, not only namespace suffix.

File: tests/Fixer/Basic/Psr4FixerTest.php [hunk 2]
```diff
@@ -29,7 +29,7 @@ public function testFixCase()
         $fileProphecy->willExtend(\SplFileInfo::class);
         $fileProphecy->getBasename('.php')->willReturn('Bar');
         $fileProphecy->getExtension()->willReturn('php');
-        $fileProphecy->getRealPath()->willReturn(__DIR__.'/Psr4/Foo/Bar.php');
+        $fileProphecy->getRealPath()->willReturn(__DIR__.\DIRECTORY_SEPARATOR.'Psr4'.\DIRECTORY_SEPARATOR.'Foo'.\DIRECTORY_SEPARATOR.'Bar.php');
```
Comment (keradus) [inline tests/Fixer/Basic/Psr4FixerTest.php:32]: is this change actually needed?
we were using `/` everywhere, and PHP under the hood were always playing fair for us, allowing for `/`and `\`as dir separator, and silence conversion under the hood for Windows paths, if needed

File: tests/Fixer/Basic/Psr4FixerTest.php (hunk 1 above)
Comment (keradus) [reply tests/Fixer/Basic/Psr4FixerTest.php:50]: Original spec of this fixer was to keep `Psr4_Foo_Bar` here - ref https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v2.11.2/tests/Fixer/Basic/Psr4FixerTest.php#L50
I would indeed keep the original version here.

We aim to fix as much, without much of code breaking.

File: tests/Fixer/Basic/Psr4FixerTest.php (hunk 2 above)
Comment (kubawerlos) [reply tests/Fixer/Basic/Psr4FixerTest.php:32]: it's needed, I'm using `realpath` function in the fixer as there are 2 paths to compare - the path of the file being fixed and path given as `dir` value in config.

File: src/Fixer/Basic/PsrAutoloadingFixer.php [hunk 3]
```diff
@@ -135,7 +135,7 @@ protected function createConfigurationDefinition()
     /**
      * {@inheritdoc}
      */
-    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
```
Comment (keradus) [inline src/Fixer/Basic/PsrAutoloadingFixer.php:138]: i think it will break on 2.x line
let's revert that and we should auto-introduce it after dropping 5.x

**Comment by keradus** [comment]
Thank you @kubawerlos.
